ui/src/app/components/runtime-configuration-panel/confirm-environment-delete-with-pd.tsx

```diff
@@ -1,29 +1,48 @@
 import * as React from 'react';
 
+import { Disk } from 'generated/fetch';
+
+import { UIAppType } from 'app/components/apps-panel/utils';
 import { Button } from 'app/components/buttons';
 import { FlexRow } from 'app/components/flex';
 import { ClrIcon } from 'app/components/icons';
 import { RadioButton } from 'app/components/inputs';
 import colors from 'app/styles/colors';
-import { ComputeType, detachableDiskPricePerMonth } from 'app/utils/machines';
+import { DEFAULT, switchCase } from 'app/utils';
+import { detachableDiskPricePerMonth } from 'app/utils/machines';
 import { formatUsd } from 'app/utils/numbers';
-import { RuntimeStatusRequest } from 'app/utils/runtime-utils';
 
 import { BackupFilesHelpSection } from './backup-files-help-section';
 import { styles } from './styles';
 
-const { useState, Fragment } = React;
+const { useState } = React;
+
+interface ConfirmDeleteEnvironmentWithPDProps {
+  onCancel: () => void;
+  onConfirm: (deletePDSelected: boolean) => Promise<void>;
+  appType: UIAppType;
+  // assumption: usingDataproc is always false for GKE apps
```

Comment:
pretty sure this is true by definition :)

Reply:
Yeah, I wanted to call out that "business logic" for this component since the component's logic depends on it.